ActivatePubKey/handler.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  retrievedLollipopKeysToApiActivatedPubKey,
+  retrievedValidPopDocument
+} from "../utils/lollipopKeys";
+import {
+  getAlgoFromAssertionRef,
+  getAllAssertionsRef
+} from "../utils/lollipopKeys";
+import { domainErrorToResponseError } from "../utils/errors";
+
+type ActivatePubKeyHandler = (
+  context: Context,
+  assertion_ref: AssertionRef,
+  body: ActivatePubKeyPayload
+) => Promise<
+  | IResponseSuccessJson<ActivatedPubKey>
+  | IResponseErrorNotFound
+  | IResponseErrorValidation
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorInternal
+>;
+
+export const ActivatePubKeyHandler = (
+  popDocumentReader: PopDocumentReader,
+  popDocumentWriter: PopDocumentWriter,
+  assertionWriter: AssertionWriter
+): ActivatePubKeyHandler => (
+  _,
+  assertion_ref,
+  body
+): ReturnType<ActivatePubKeyHandler> =>
+  pipe(
+    popDocumentReader(assertion_ref),
+    TE.mapLeft(domainErrorToResponseError),
+    TE.filterOrElseW(PendingLolliPopPubKeys.is, () =>
+      ResponseErrorInternal("Unexpected status on pop document")
+    ),
+    TE.chainW(popDocument =>
+      pipe(
+        `${body.fiscal_code}-${assertion_ref}`,
+        AssertionFileName.decode,
+        TE.fromEither,
+        TE.bindTo("assertionFileName"),
+        TE.mapLeft(errors =>
+          ResponseErrorInternal(
+            `Could not decode assertionFileName | ${readableReport(errors)}`
+          )
+        ),
+        TE.chainFirst(({ assertionFileName }) =>
+          pipe(
+            assertionWriter(assertionFileName, body.assertion),
+            TE.mapLeft(error => ResponseErrorInternal(error.detail))
+          )
+        ),
+        TE.bind("jwkPubKeyFromString", () =>
+          pipe(
+            popDocument.pubKey,
+            JwkPublicKeyFromToken.decode,
+            TE.fromEither,
+            TE.mapLeft(errors =>
+              ResponseErrorInternal(
+                `Could not decode public key | ${readableReport(errors)}`
+              )
+            )
+          )
+        ),
+        TE.bind("assertionRefs", ({ jwkPubKeyFromString }) =>
+          pipe(
+            getAllAssertionsRef(
+              JwkPubKeyHashAlgorithmEnum.sha512,
+              getAlgoFromAssertionRef(popDocument.assertionRef),
+              jwkPubKeyFromString
+            ),
+            TE.mapLeft((error: Error) => ResponseErrorInternal(error.message))
+          )
+        ),
+        TE.bind(
+          "retrievedPopDocument",
+          ({ assertionRefs, assertionFileName }) =>
+            pipe(
+              popDocumentWriter({
+                assertionFileName,
+                assertionRef: assertionRefs.master,
+                assertionType: body.assertion_type,
+                expiredAt: body.expires_at,
+                fiscalCode: body.fiscal_code,
+                pubKey: popDocument.pubKey,
+                status: PubKeyStatusEnum.VALID,
+                ttl: TTL_VALUE_AFTER_UPDATE
```

Comment:
add comment on why this value will be unused

Reply:
Made ttl partial for `NewLolliPopPubKeys`type  in [d450057](https://github.com/pagopa/io-functions-lollipop/pull/13/commits/d4500570e87494922f1c25cc1197a1a58287cc75)